tests/framework/e2e/downgrade.go

```diff
@@ -117,6 +131,14 @@ func DowngradeUpgradeMembersByID(t *testing.T, lg *zap.Logger, clus *EtcdProcess
 			return err
 		}
 	}
+
+	if opString == "downgrading" && len(membersToChange) == len(clus.Procs) {
+		lg.Info("Waiting for downgrade completion log line")
+		leader := clus.WaitLeader(t)
+		_, err := clus.Procs[leader].Logs().ExpectWithContext(context.TODO(), expect.ExpectedResponse{Value: "the cluster has been downgraded"})
```

Comment:
Please set a timeout, i.e. 30s.

Reply:
Done as requested!

Do you see a risk with this approach of monitoring the leader's log @ahrtr? :)

Reply:
The only concern is it will fail if the leader somehow changes.

So suggest to add a retry mechanism (i.e. up to 3 times), and let's wait 15s instead of 30s in each retry.

Reply:
Done as suggested